### PR TITLE
[Mesheryctl] Added list to context

### LINKED ISSUE #2708: [mesheryctl] `system context list` to list all the contexts
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior

`mesheryctl system context view` will show the config for the `current-context` and `mesheryctl system context view --context <context name>` will show the config for the context `<context name>`.

#### Desired Behavior

However, users cannot view all the available contexts from the CLI and would have to manually check the `~/.meshery/config.yaml`.

A new subcommand `mesheryctl system context list` that will list the `current-context` and all the available contexts.

```
❯ mesheryctl system context list

Current context: prod-east-1

Available contexts:

local
prod-east-1
dev
```

Users will then be able to use this context name to view the config of the context- `mesheryctl system context view --context dev`.

#### Contributor Resources

- [mesheryctl Contributing Guide](https://github.com/layer5io/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)

### PR DESCRIPTION
**Pull request by Kn99HN**
**Description**
- Added `list` as subcommand to `context` to show current and all available contexts.

This PR fixes #2708 

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@Kn99HN That was a fast fix! You have some lint errors that would need some tidying up.

**Comment by Kn99HN** [comment]
> @Kn99HN That was a fast fix! You have some lint errors that would need some tidying up.

Thanks! Is the project using gofmt? I didn't quite catch the linter for Go in the `Contributing.md`. I'm a bit new to Go so I just want to make sure

**Comment by navendu-pottekkat** [comment]
We use this linter https://github.com/golangci/golangci-lint

The GitHub action provides quite useful info on what the error is actually-

Screenshot from 2021-04-14 17-51-38

**Comment by leecalcote** [comment]
Am I mistaken or was this PR swiftly delivered? It’s great to see you move though this, @Kn99HN. 

The linter used in these builds is expecting that golang files have been formatted with`gofmt -s`.

File: mesheryctl/internal/cli/root/system/context/list.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		log.Print("Available contexts:\n")
+		for context := range contexts {
+			log.Printf("%s", context)
+		}
+		return nil
+	},
+}
+
+func init() {
+	listContextCmd.Flags().StringVar(&currContext, "context", "", "Show config for the context")
```
Comment (navendu-pottekkat) [inline mesheryctl/internal/cli/root/system/context/list.go:45]: We would not need a flag here since `context` is a global flag for all system commands.

**Review by navendu-pottekkat — COMMENTED** [review]
@Kn99HN I have left a couple of suggestions to look through... The rest looks good.

File: mesheryctl/internal/cli/root/system/context/list.go (hunk 1 above)
Comment (Kn99HN) [reply mesheryctl/internal/cli/root/system/context/list.go]: Removed! Thanks for pointing it out

File: mesheryctl/internal/cli/root/system/context/list.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		var contexts = configuration.Contexts
+		if contexts == nil {
+			return errors.New("no contexts set")
+		}
+
+		if currContext == "" {
+			currContext = viper.GetString("current-context")
+		}
+		if currContext == "" {
+			return errors.New("current context not set")
```
[the remark this answers is not in the document]
Comment (Kn99HN) [reply mesheryctl/internal/cli/root/system/context/list.go]: Yep, I changed the output for this.

File: mesheryctl/internal/cli/root/system/context/list.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		} else {
+			log.Printf("Current context: %s\n", currContext)
+		}
+		log.Print("Available contexts:\n")
+		for context := range contexts {
+			log.Printf("- %s", context)
+		}
+
+		if currContext == "" {
+			log.Print("\nRun to set current context: mesheryctl system context switch <context name>")
```
Comment (navendu-pottekkat) [inline mesheryctl/internal/cli/root/system/context/list.go:43]:
```suggestion
			log.Print("\nRun `mesheryctl system context switch <context name>` to set the current context.")
```

**Review by navendu-pottekkat — APPROVED** [review]
LGTM. Thank you @Kn99HN

**Pull request merged**